assets/js/amp-customize-controls.js

```diff
@@ -0,0 +1,194 @@
+			if ( ampToggle && ! ampEnabled ) {
+				toggleControl( false, true, true );
+			} else if ( ! ampToggle && ! ampEnabled ) {
+
+				// Enable toggle but don't trigger anything else.
+				$( '.amp-toggle input' ).prop( 'disabled', false );
+			}
+		}
+
+		/**
+		 * Central function to control UI toggle elements.
+		 *
+		 * @param {bool} checked If toggle input should be checked.
+		 * @param {bool} disabled If toggle input should be disabled.
+		 * @param {bool} notification If tooltip notification should appear.
+		 * @return {void}
+		 */
+		function toggleControl( checked, disabled, notification ) {
+			var $input = $( '.amp-toggle input' ),
+				$tooltip = $( '.amp-toggle .tooltip' ),
+				tooltipTimer = 5000;
+
+			$input.prop( 'checked', checked );
+			$input.prop( 'disabled', disabled );
+
+			if ( notification ) {
+				$tooltip.fadeIn();
+				setTimeout( function() {
+					$tooltip.fadeOut();
+				}, tooltipTimer );
+			} else {
+				$tooltip.hide();
+			}
+
+			$input.trigger( 'change' );
+		}
+
+		// AMP panel triggers the input toggle for AMP preview.
+		panel.expanded.bind( function() {
+			var panelOpen = panel.expanded.get();
+			toggleControl( panelOpen, panelOpen, false );
+		} );
+
+		// If AMP panel is open, let it handle enable/disable of the toggle,
+		// otherwise auto enable toggle when mobile device is selected.
+		api.previewedDevice.bind( function( device ) {
+			if ( ! panel.expanded.get() && ! ampToggle ) {
+				toggleControl( 'mobile' === device, false, false );
+			}
+		} );
+
+		// Preview is ready, check if it's AMP'd in case we need to show tooltip.
+		api.previewer.bind( 'ready', isPageAmpified );
+
+		// Persist the presence or lack of the amp=1 param when navigating in the preview,
+		// even if current page is not yet supported.
+		api.previewer.previewUrl.validate = ( function( prevValidate ) {
+			return function( value ) {
+				var val = prevValidate.call( this, value );
+				if ( val ) {
+					val = setCurrentAmpUrl( val );
+				}
+				return val;
+			};
+		} )( api.previewer.previewUrl.validate );
+
+		// Adding checkbox toggle before device selection.
+		$( '.devices-wrapper' ).before( '<label class="amp-toggle">' +
+			'<input type="checkbox">' +
+			'<span class="slider"></span>' +
+		'</label>' );
+
+		// Notification tooltip for AMP toggle.
+		$( '.amp-toggle input' ).before( '<span class="tooltip">' +
+			'This page is not AMP compatible.<br>' +
+			'<a data-post="' + window.ampPost + '">Navigate to an AMP compatible page</a>' +
+		'</span>' );
+
+		$( '.amp-toggle .tooltip a' ).on( 'click', function() {
+			var url = $( this ).data( 'post' );
+			if ( url.length ) {
+				api.previewer.previewUrl.set( setCurrentAmpUrl( url ) );
+				toggleControl( false, false, false );
+			}
+		} );
+
+		// Main control for toggling AMP preview.
+		$( '#customize-footer-actions' ).on( 'change', '.amp-toggle input', function( event ) {
+			ampToggle = $( this ).is( ':checked' );
+			toggleAmpView();
```

Comment:
I don't think the toggle should be toggle-able if the currently-previewed URL is not AMP-compatible. It should be `disabled` in that case, where clicking it just then shows the tooltip.